packages/esm-patient-programs-app/src/programs/delete-program.modal.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback, useState } from 'react';
+import { Button, InlineLoading, ModalBody, ModalFooter, ModalHeader } from '@carbon/react';
+import { useTranslation } from 'react-i18next';
+import { deleteProgramEnrollment, useEnrollments } from './programs.resource';
+import { showSnackbar, getCoreTranslation } from '@openmrs/esm-framework';
+
+interface DeleteProgramProps {
+  closeDeleteModal: () => void;
+  programEnrollmentId: string;
+  patientUuid: string;
+}
+
+const DeleteProgramModal: React.FC<DeleteProgramProps> = ({ closeDeleteModal, programEnrollmentId, patientUuid }) => {
+  const { t } = useTranslation();
+  const [isDeleting, setIsDeleting] = useState(false);
+  const { mutateEnrollments } = useEnrollments(patientUuid);
+
+  const handleDelete = useCallback(async () => {
+    try {
+      setIsDeleting(true);
+      const response = await deleteProgramEnrollment(programEnrollmentId, new AbortController());
+      if (response.ok) {
+        mutateEnrollments();
+        closeDeleteModal();
+        showSnackbar({
+          isLowContrast: true,
+          kind: 'success',
+          title: t('programDeleted', 'Program Enrollment Deleted'),
+        });
+      }
```

Comment:
```suggestion
      await mutateEnrollments();
      closeDeleteModal();
      
      showSnackbar({
        isLowContrast: true,
        kind: 'success',
        title: t('programDeleted', 'Program Enrollment Deleted'),
      });
```

`mutateEnrollments` is a promise.